src/components/helpers/filterDefinitions.js

```diff
@@ -0,0 +1,215 @@
+import { Types } from '@/helpers/typeHelper';
+
+export const legacyTypes = {
+  [Types.directories]: [],
+  [Types.text]: [
+    // eBook types
+    'application/x-mobipocket-ebook',
+    'application/epub+zip',
+    'application/vnd.amazon.ebook',
+    // Scanned documents
+    'image/vnd.djvu',
+    'application/pdf',
+    // HTML/plain text
+    'text/html',
+    'text/plain',
+    // Text editors
+    'application/postscript',
+    'application/rtf',
+    // Open Office et al.
+    'application/vnd.oasis.opendocument.text',
+    'application/vnd.sun.xml.writer',
+    'application/vnd.stardivision.writer',
+    'application/x-starwriter',
+    // MS Word
+    'application/msword',
+    'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+    // Misc
+    'application/x-abiword',
+  ],
+  [Types.audio]: [
+    'audio*',
+    // 'application/ogg',
+  ],
+  [Types.video]: [
+    'video*',
+    // 'application/mp4'
+  ],
+  [Types.images]: [
+    'image*',
+  ],
+};
+
+export const extensions = {
```

Comment:
I don't understand the relevance of extensions here if we have mimetypes defined, as well as an API/library for rendering those mimetypes as extensions for user-convenience. I would strongly recommend to filter based on mimetypes over extensions, also because most content has a mimetype while many files do not have even a filename let alone an extension defined.

I would also suggest to make the filtering API rich/expressive enough to:
1. Allow search results to define available type filters (so the list of types can be based on actual types in a search result).
2. Allow search queries to define available filters (e.g. directories won't have type filters, documents have language filters based on return languages etc.).

So then the list of filters would be a responsive thingy based on results and query from the store. Does that make sense?

Reply:
We might have to be careful to discriminate 'subtypes' (e.g. PDF, Word, ePub, MP3, MKV) from 'types' (e.g. Audio, Documents, Video) in the UX.

Perhaps best to discuss this with @vandrongelen.

Reply:
yeah i agree this needs some more thought. 
It should be trivial to turn the extension filter into a mimetype filter.

Reply:
> I don't understand the relevance of extensions here if we have mimetypes defined, as well as an API/library for rendering those mimetypes as extensions for user-convenience. I would strongly recommend to filter based on mimetypes over extensions, also because most content has a mimetype while many files do not have even a filename let alone an extension defined.

It might be better to make a mimetype-subfilter rather than extension filter indeed; the extension filter is not very good, because not all files have extensions. It is very easy to change the filter definitions to make it so.

> I would also suggest to make the filtering API rich/expressive enough to:
> 
> 1. Allow search results to define available type filters (so the list of types can be based on actual types in a search result).
> 2. Allow search queries to define available filters (e.g. directories won't have type filters, documents have language filters based on return languages etc.).
> 
> So then the list of filters would be a responsive thingy based on results and query from the store. Does that make sense?

1. Makes no sense. You can't prefilter the query based on its results. If you want to make post-query filter, this means rethinking the UX of it all. Currently, the filters update the query. 
2. This is already happening. If you select directories, there is no extension filter. The same mechanism can be used for other types of filters, e.g., if you wanted to make a language filter only for text documents.

Reply:
Alright, so let's remove extenion filters, or refactor them to use mimetypes.

Reply:
Also:

1. It seems that currently extension filters *are* disabled but I have not been able to discover where or how in the code this is done. I really need to know how to add more filters and I find the current code/API design *very* hard to understand. 

2. The current filter values do seem static, that is, they cannot arbitrarily change based on user-defined conditions

Couldn't it be as single as a `getFilters()`, which returns a bunch of queries (based on the query and the results from their respective stores)? The filters format might return a list of filters very similar to what `filterDefinitions.js` returns, although perhaps a slug-based hash makes composition more elegant and legible.

Note that making assumptions about types in the *architecture* if the filter infrastructure really seems an anti-feature, there is no guarantee that categories like `directory` and `text` will be maintained, or the concept of type categories in general for that matter. Plus it doesn't seem necessary to have this concept.

It seems to me it could be as simple as (pseudo-code based on imagination API):

```javascript
function getLanguageFilter() {
	let languages = getValuesFromResultsStore('language');
	...
}

function getSubtypeFilter() {
	let mimetypes = getValuesFromResultsStore('mimetypes');
	...

}

function getAnyFilters() {
	return {
		size: getSizeFilter(),
		...
	}

}

function getDocumentFilters() {
	return {
		subtype: getSubtypeFilter(),
		language: getLanguageFilter(),
		...
	}

}

function getFilters() {
	let type = getFilterFromQueryStore('type');

	switch (type) {
		case 'any':
			return getAnyFilters();
		case 'audio':
			return getDocumentFilters();
		...
	}

}
```